Common/UI/SkillsTree/SkillSelectionPanel.cs

```diff
@@ -0,0 +1,65 @@
+﻿using PathOfTerraria.Common.Mechanics;
+using PathOfTerraria.Core.UI.SmartUI;
+using Terraria.ModLoader.Core;
+
+namespace PathOfTerraria.Common.UI.SkillsTree;
+
+internal class SkillSelectionPanel : SmartUiElement
+{
+	private SkillTreeInnerPanel _skillTreeInnerPanel;
+	public override string TabName => "SkillTree";
+
+	private bool _drewSkills;
+	public Skill SelectedSkill { get; set; }
+	
+	public override void Draw(SpriteBatch spriteBatch)
+	{
+		base.Draw(spriteBatch);
+		if (!_drewSkills && SelectedSkill == null)
+		{
+			_drewSkills = true;
+			AppendAllSkills();
+		}
+	}
+
+	private void AppendAllSkills()
+	{
+		int index = 0;
+		foreach (Type type in AssemblyManager.GetLoadableTypes(PoTMod.Instance.Code))
+		{
+			if (type.IsAbstract || !type.IsSubclassOf(typeof(Skill)))
+			{
+				continue;
+			}
+
+			var skill = (Skill) Activator.CreateInstance(type);
+			var element = new SkillSelectionElement(skill, index, this);
+			Append(element);
+			index += 1;
+		}
+	}
+
+	public void DrawSkillTree()
```

Comment:
When naming methods, "Draw...()" conventionally means a method that is called every frame and draws stuff. This is not remotely what this method does. Something like "RebuildTree" would be more appropriate.